packages/skill-lesson/hooks/use-golden-ticket.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from 'react'
+import {useRouter} from 'next/router'
+import {useCoupon} from '../path-to-purchase/use-coupon'
+import {trpcSkillLessons} from '../utils/trpc-skill-lessons'
+import {Decimal} from '@skillrecordings/database'
+import * as _skillrecordings_database from '@skillrecordings/database'
+
+export type GoldenTicketContextType = {
+  redeemableCoupon?: boolean
+  RedeemDialogForCoupon: () => React.JSX.Element | null
+  validCoupon: boolean
+  invalidReason: string
+  couponData?:
+    | {
+        isValid: boolean
+        isRedeemable: boolean
+        error: string
+        isExpired?: undefined
+        isUsedUp?: undefined
+        id: string
+        code: string | null
+        createdAt: Date
+        expires: Date | null
+        maxUses: number
+        default: boolean
+        merchantCouponId: string | null
+        status: number
+        usedCount: number
+        percentageDiscount: Decimal
+        restrictedToProductId: string | null
+        bulkPurchaseId: string | null
+      }
+    | {
+        isExpired: boolean
+        isUsedUp: boolean
+        isRedeemable: boolean
+        isValid: boolean
+        error?: undefined
+        id: string
+        code: string | null
+        createdAt: Date
+        expires: Date | null
+        maxUses: number
+        default: boolean
+        merchantCouponId: string | null
+        status: number
+        usedCount: number
+        percentageDiscount: Decimal
+        restrictedToProductId: string | null
+        bulkPurchaseId: string | null
+      }
+    | null
+}
```

Comment:
this type is generated by prisma and then augmented so I cut and pasted it. Probably a bad approach...

Reply:
made it make sense